bundles/org.openhab.core.io.rest.core/src/main/java/org/openhab/core/io/rest/core/internal/item/ItemResource.java

```diff
@@ -756,6 +758,34 @@ public Response createOrUpdateItems(
         return JSONResponse.createResponse(Status.OK, responseList, null);
     }
 
+    @GET
+    @RolesAllowed({ Role.USER, Role.ADMIN })
+    @Path("/{itemName: \\w+}/semantic/{semanticClass: \\w+}")
+    @Operation(operationId = "getSemanticItem", summary = "Gets the item which defines the requested semantics of an item.", responses = {
+            @ApiResponse(responseCode = "200", description = "OK"),
+            @ApiResponse(responseCode = "404", description = "Item not found") })
+    public Response getSemanticItem(final @Context UriInfo uriInfo, final @Context HttpHeaders httpHeaders,
+            @HeaderParam(HttpHeaders.ACCEPT_LANGUAGE) @Parameter(description = "language") @Nullable String language,
+            @PathParam("itemName") @Parameter(description = "item name") String itemName,
+            @PathParam("semanticClass") @Parameter(description = "semantic class") String semanticClassName) {
+        Locale locale = localeService.getLocale(language);
+
+        Class<? extends org.openhab.core.semantics.Tag> semanticClass = SemanticTags.getById(semanticClassName);
+        if (semanticClass == null) {
+            return Response.status(Status.NOT_FOUND.getStatusCode()).build();
```

Comment:
Isn't it sufficient to pass the status itself?

```suggestion
            return Response.status(Status.NOT_FOUND).build();
```